src/base/bittorrent/trackerentry.cpp

```diff
@@ -94,8 +116,10 @@ int TrackerEntry::numSeeds() const
 #if (LIBTORRENT_VERSION_NUM < 10200)
     return nativeEntry().scrape_complete;
 #else
-    // FIXME: Handle all possible endpoints.
-    return nativeEntry().endpoints.empty() ? -1 : nativeEntry().endpoints[0].scrape_complete;
+    int max = -1;
+    for (const lt::announce_endpoint &endpoint : nativeEntry().endpoints)
```

Comment:
Maybe use std::max_element?

Reply:
```c++
const auto &endpoints = nativeEntry().endpoints;
const auto maxElement = std::max_element(endpoints.cbegin(), endpoints.cend()
    , [](const lt::announce_endpoint &left, const lt::announce_endpoint &right)
{
    return (left.scrape_complete < right.scrape_complete);
});
return (maxElement == endpoints.cend()) ? -1 : maxElement->scrape_complete;
```
Looks horrible, I somehow don't prefer it.

Or another version:
```c++
const auto &endpoints = nativeEntry().endpoints;
return std::accumulate(endpoints.cbegin(), endpoints.cend(), -1
    , [](const int acc, const lt::announce_endpoint &endpoint)
{
    return std::max(acc, endpoint.scrape_complete);
});
```
A bit less readable than the simple for-loop I think.

Reply:
>Looks horrible

I meant:
```c++
const auto &endpoints = nativeEntry().endpoints;
return endpoints.empty() ? -1 : *std::max_element(endpoints.cbegin(), endpoints.cend()
    , [](const lt::announce_endpoint &left, const lt::announce_endpoint &right)
{
    return (left.scrape_complete < right.scrape_complete);
});
```
But you can leave it as is if it is horrible too for you.

Reply:
I'll keep the simple for-loop for now.